File: src/main/java/com/chaottic/nbt/number/ByteTag.java
package com.chaottic.nbt.number;

import java.io.DataInput;
import java.io.IOException;

public record ByteTag(byte b) implements NumberTag {
    @Override
    public byte getAsByte() {
        return b;
    }

    @Override
    public short getAsShort() {
        return b;
    }

    @Override
    public int getAsInt() {
        return b;
    }

    @Override
    public long getAsLong() {
        return b;
    }

    @Override
    public float getAsFloat() {
        return b;
    }

    @Override
    public double getAsDouble() {
        return b;
    }

    public static ByteTag createByteTag(DataInput dataInput) throws IOException {
        return new ByteTag(dataInput.readByte());
    }
}

File: src/main/kotlin/com/chaottic/nbt/number/ByteTag.kt
package com.chaottic.nbt.number

import com.chaottic.nbt.TagType
import java.io.DataInput

@JvmInline
value class ByteTag(private val b: Byte) : NumberTag {

	override fun asByte() = b
	override fun asShort() = b.toShort()
	override fun asInt() = b.toInt()
	override fun asLong() = b.toLong()
	override fun asFloat() = b.toFloat()
	override fun asDouble() = b.toDouble()

	companion object : TagType<ByteTag> {

		override fun createTag(dataInput: DataInput) = ByteTag(dataInput.readByte())
	}
}
